labstep/__init__.py

```diff
@@ -1,581 +1,646 @@
-
+    Parameters
+    ----------
+    user (obj)
+        The Labstep user whose Resources you want to retrieve.
+        Must have property 'api_key'. See 'login'. 
+    count (int)
+        The number of Resources to retrieve. 
+
+    Returns
+    -------
+    resources
+        A list of Resource objects.
+    '''
+    headers = {'apikey': user['api_key']}  
+    n = min(count,1000)
+    url = "https://api.labstep.com/api/generic/resource"
+    params = {'search': 1,
+              'cursor':-1,
+              'count':n}
+    r = requests.get(url, params=params, headers=headers)
+    resp = json.loads(r.content)
+    items = resp['items']
+
+    expected_results = min(resp['total'],count)
+    while len(items)<expected_results:
+        cursor = resp['next_cursor']
+        params = {'search':1,
+                  'cursor':cursor,
+                  'count':n}
+        r = requests.get(url, params=params, headers=headers)
+        resp = json.loads(r.content)
+        items.extend(resp['items'])
+    return items
 
 def getProtocols(user,count=100):
-  '''
-  Retrieve a list of a users Protocols on Labstep.
+    '''
+    Retrieve a list of a user's Protocols on Labstep.
   
-  Parameters
-  ----------
-  user : obj
-    The Labstep user whose Protocols you want to retrieve. Must have property 'api_key'. See 'login'. 
-  count : int
-    The number of Protocols to retrieve. 
-
-  Returns
-  -------
-  protocols
-    A list of Protocol objects.
-  '''
-  headers = {'apikey': user['api_key']}
-  n = min(count,1000)
-  playlist_id = user['protocol_collection_playlists'][0]['id']
-  url = "https://api.labstep.com/api/generic/protocol-collection"
-  params = {'search':1,
-            'cursor':-1,
-            'count':n,
-            'protocol_collection_playlist_id':playlist_id
-            }
-  r = requests.get(url, params=params, headers=headers)
-
-  resp = json.loads(r.content)
-  items = resp['items']
-
-  expected_results = min(resp['total'],count)
-
-  while len(items)<expected_results:
+    Parameters
+    ----------
+    user (obj)
+        The Labstep user whose Protocols you want to retrieve.
+        Must have property 'api_key'. See 'login'. 
+    count (int)
+        The number of Protocols to retrieve. 
+
+    Returns
+    -------
+    protocols
+        A list of Protocol objects.
+    '''
+    headers = {'apikey': user['api_key']}
+    n = min(count,1000)
+    url = "https://api.labstep.com/api/generic/protocol-collection"
     params = {'search':1,
-              'cursor':resp['next_cursor'],
-              'count':n,
-              'protocol_collection_playlist_id':playlist_id
-              } 
+              'cursor':-1,
+              'count':n}
     r = requests.get(url, params=params, headers=headers)
-
     resp = json.loads(r.content)
-    items.extend(resp['items'])
-
-  return items
-
-
+    items = resp['items']
+
+    expected_results = min(resp['total'],count)
+    while len(items)<expected_results:
```

Comment:
We reuse this pattern in all three of the get[X]s functions. Can you think of a way to abstract it out to a function like 'keepGetting' which takes the url, the params and the number requested and implements this loop? Then each of these get[X]s can just call keepGetting and will be a lot cleaner.